src/components/Settings/SettingsUsers/index.tsx

```diff
@@ -42,6 +51,31 @@ const SettingsUsers = () => {
   } = useSWR<MainSettings>('/api/v1/settings/main');
   const settings = useSettings();
 
+  const schema = useMemo(
+    () =>
+      yup
+        .object()
+        .shape({
+          localLogin: yup.boolean(),
+          mediaServerLogin: yup.boolean(),
+        })
+        .test({
+          name: 'atLeastOneAuth',
+          test: function (values) {
+            const isValid = ['localLogin', 'mediaServerLogin'].some(
+              (field) => !!values[field]
+            );
+
+            if (isValid) return true;
+            return this.createError({
+              path: 'localLogin | mediaServerLogin',
+              message: intl.formatMessage(messages.atLeastOneAuth),
+            });
+          },
+        }),
+    [intl]
+  );
```

Comment:
We a useMemo? It feels a bit unnecessary to memoize this.
```suggestion
  const schema = yup
    .object()
    .shape({
      localLogin: yup.boolean(),
      mediaServerLogin: yup.boolean(),
    })
    .test({
      name: 'atLeastOneAuth',
      test: function (values) {
        const isValid = ['localLogin', 'mediaServerLogin'].some(
          (field) => !!values[field]
        );

        if (isValid) return true;
        return this.createError({
          path: 'localLogin | mediaServerLogin',
          message: intl.formatMessage(messages.atLeastOneAuth),
        });
      },
    });
```

Reply:
It feels a bit wasteful to rebuild the schema each time the component re-renders, since it's only dependent on the intl so really only needs to be built once. But it would probably be fine without.

Reply:
I used to use memo a lot some time ago, but I've since read several articles that stated that abusing memoization is in fact counterproductive, hence my comment.
And these optimizations should (hopefully) be automatically achieved soon, with React Compiler.

Reply:
But you can also leave it like that, it's not really a problem, i mainly said that because `yup.schema()` are also not memoized in the rest of the app

Reply:
I went ahead and removed it so I think I'll just leave it like that. Hopefully React compiler will come soon 🙏